tinysegmenter/tinysegmenter.py

```diff
@@ -12,181 +12,227 @@
+_UW5 = {",":465,".":-299,"1":-514,"E2":-32768,"]":-2762,"、":465,"。":-299,"「":363,"あ":1655,"い":331,"う":-503,"え":1199,"お":527,"か":647,"が":-421,"き":1624,"ぎ":1971,"く":312,"げ":-983,"さ":-1537,"し":-1371,"す":-852,"だ":-1186,"ち":1093,"っ":52,"つ":921,"て":-18,"で":-850,"と":-127,"ど":1682,"な":-787,"に":-1224,"の":-635,"は":-578,"べ":1001,"み":502,"め":865,"ゃ":3350,"ょ":854,"り":-208,"る":429,"れ":504,"わ":419,"を":-1264,"ん":327,"イ":241,"ル":451,"ン":-343,"中":-871,"京":722,"会":-1153,"党":-654,"務":3519,"区":-901,"告":848,"員":2104,"大":-1296,"学":-548,"定":1785,"嵐":-1304,"市":-2991,"席":921,"年":1763,"思":872,"所":-814,"挙":1618,"新":-1682,"日":218,"月":-4353,"査":932,"格":1356,"機":-1508,"氏":-1347,"田":240,"町":-3912,"的":-3149,"相":1319,"省":-1052,"県":-4003,"研":-997,"社":-278,"空":-813,"統":1955,"者":-2233,"表":663,"語":-1073,"議":1219,"選":-1018,"郎":-368,"長":786,"間":1191,"題":2368,"館":-689,"１":-514,"Ｅ２":-32768,"｢":363,"ｲ":241,"ﾙ":451,"ﾝ":-343};
+_UW6 = {",":227,".":808,"1":-270,"E1":306,"、":227,"。":808,"あ":-307,"う":189,"か":241,"が":-73,"く":-121,"こ":-200,"じ":1782,"す":383,"た":-428,"っ":573,"て":-1014,"で":101,"と":-105,"な":-253,"に":-149,"の":-417,"は":-236,"も":-206,"り":187,"る":-135,"を":195,"ル":-673,"ン":-496,"一":-277,"中":201,"件":-800,"会":624,"前":302,"区":1792,"員":-1212,"委":798,"学":-960,"市":887,"広":-695,"後":535,"業":-697,"相":753,"社":-507,"福":974,"空":-822,"者":1811,"連":463,"郎":1082,"１":-270,"Ｅ１":306,"ﾙ":-673,"ﾝ":-496};
+
+def _ctype(c):
+    return _CHARDICT.get(c, 'O')
+
+def tokenize(text):
+    if text == "":
+        return []
+
+    result = []
+    seg = ["B3","B2","B1"]
+    ctype = ["O","O","O"]
+    seg.extend(text)
+    ctype.extend(map(_ctype, text))
+
+    seg.append("E1")
+    seg.append("E2")
+    seg.append("E3")
+    ctype.append("O")
+    ctype.append("O")
+    ctype.append("O")
+
+    word = seg[3]
+    p1 = "U"
+    p2 = "U"
+    p3 = "U"
+
+    # method cache
+    up1 = _UP1.get
+    up2 = _UP2.get
+    up3 = _UP3.get
+    bp1 = _BP1.get
+    bp2 = _BP2.get
+    uw1 = _UW1.get
+    uw2 = _UW2.get
+    uw3 = _UW3.get
+    uw4 = _UW4.get
+    uw5 = _UW5.get
+    uw6 = _UW6.get
+    bw1 = _BW1.get
+    bw2 = _BW2.get
+    bw3 = _BW3.get
+    tw1 = _TW1.get
+    tw2 = _TW2.get
+    tw3 = _TW3.get
+    tw4 = _TW4.get
+    uc1 = _UC1.get
+    uc2 = _UC2.get
+    uc3 = _UC3.get
+    uc4 = _UC4.get
+    uc5 = _UC5.get
+    uc6 = _UC6.get
+    bc1 = _BC1.get
+    bc2 = _BC2.get
+    bc3 = _BC3.get
+    tc1 = _TC1.get
+    tc2 = _TC2.get
+    tc3 = _TC3.get
+    tc4 = _TC4.get
+    # tc5 = _TC5.get
+    uq1 = _UQ1.get
+    uq2 = _UQ2.get
+    uq3 = _UQ3.get
+    bq1 = _BQ1.get
+    bq2 = _BQ2.get
+    bq3 = _BQ3.get
+    bq4 = _BQ4.get
+    tq1 = _TQ1.get
+    tq2 = _TQ2.get
+    tq3 = _TQ3.get
+    tq4 = _TQ4.get
+
+    for i in xrange(4, len(seg) - 3):
```

Comment:
hard to understand why you need to use `-3`.
Is there any better way to express this statement ?